src/components/instructor/AssignmentUpdate.js

```diff
@@ -1,16 +1,80 @@
 import React, { useState } from 'react';
+import Button from "@mui/material/Button";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import TextField from "@mui/material/TextField";
+import DialogActions from "@mui/material/DialogActions";
+import {SERVER_URL} from "../../Constants";
 
 //  instructor updates assignment title, dueDate 
 //  use an mui Dialog
 //  issue PUT to URL  /assignments with updated assignment
 
 const AssignmentUpdate = (props)  => {
+  const [open, setOpen] = useState(false);
+  const [editMessage, setEditMessage] = useState('');
+  const [assignment, setAssignment] = useState(
+      {id: '', title: '', dueDate: '', secId: '', secNo: ''});
 
+  const editOpen = (event) => {
+    setOpen(true);
+    setEditMessage('');
+    setAssignment(props.asgnmts);
+  };
 
+  const editClose = () => {
+    setOpen(false);
+    setAssignment({id: '', title: '', dueDate: '', secId: '', secNo: ''});
+
+  };
+
+  const editChange = (event) => {
+    setAssignment({...assignment,  [event.target.name]:event.target.value})
+  }
+
+  const onSave = () => {
+    saveSection(assignment);
+  }
+
+  const saveSection = async (assignment) => {
+    try {
+      const response = await fetch (`${SERVER_URL}/assignments`,
+          {
+            method: 'PUT',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(assignment),
+          });
+      if (response.ok) {
+        setEditMessage("section saved");
+        editClose();
```

Comment:
delete `editClose();` here. this will cause an error with the async function i believe.